rules/ast/ast.go

```diff
@@ -29,7 +29,12 @@ import (
 	"github.com/prometheus/prometheus/storage/metric"
 )
 
-var stalenessDelta = flag.Duration("query.staleness-delta", 300*time.Second, "Staleness delta allowance during expression evaluations.")
+var (
+	stalenessDelta = flag.Duration("query.staleness-delta", 300*time.Second, "Staleness delta allowance during expression evaluations.")
+	queryTimeout = flag.Duration("query.timeout", 30*time.Second, "Maximum time a query may take before being aborted.")
+)
+
+var QueryTimeoutErr = errors.New("Query timeout")
```

Comment:
Since we don't foresee using this name outside of this package, let's lower-case `queryTimeoutErr`.

Some other conventions: error texts always begin lower-case (see https://code.google.com/p/go-wiki/wiki/CodeReviewComments#Error_Strings), and since the user will actually get to see this message, how about making it a bit more verbose to make it clear exactly after how much time queries time out:

    var queryTimeoutErr = fmt.Errorf("query timed out after %v", *queryTimeout)